.rubocop.yml

```diff
@@ -154,6 +154,8 @@ Style/HashExcept:
   Enabled: true
 Style/HashLikeCase:
   Enabled: false
+Style/HashSyntax:
+  EnforcedShorthandSyntax: either
```

Comment:
This allows but doesn't require the new shorthand syntax. See detailed notes in the relevant commit message.

Reply:
Yeah this makes sense to me. I'm not quite sure how I feel about the new syntax, in js/ts I think it's fine so probably just takes a bit of getting used to. Either way I probably would avoid major syntax style changes in this upgrade, and once everything works, re-evaluate on that stuff

Reply:
@jurre good thought on separating out the major syntax style changes, I added a commit temporarily doing that for some freezing style changes... that keeps this PR's diff very focused.